components/BackendGamification.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+  <BackendProfile ref="backendProfile" />
+</template>
+
+<script setup lang="jsx">
+const backendProfile = ref(null);
+
+// Initialise points and score
+const streak = ref({});
+
+onMounted(() => {
+  backendProfile;
+  getStreak();
+});
+
+function getStreak() {
+  if (typeof window !== "undefined") {
+    const storedStreak = localStorage.getItem("streak");
+    if (!storedStreak) {
+      streak.value = { streak: 1, lastDate: "" };
+      localStorage.setItem("streak", JSON.stringify(streak.value));
+    } else {
+      streak.value = storedStreak;
+      // TODO: Consider adding logic to reset streak if lastDate is not today
+    }
+  }
+  return streak.value;
+}
+
+function addStreak() {
+  const storedStreak = JSON.parse(getStreak()).streak;
+  const currentDate = new Date();
+  const formattedDate = currentDate.toISOString().slice(0, 10);
+
+  streak.value = {
+    streak: storedStreak + 1,
+    lastDate: formattedDate,
+  };
+
+  // TODO: Add highest streak logic here
+  localStorage.setItem("streak", JSON.stringify(streak.value));
+}
+
+/**
+ * Get the points from localStorage
+ *
+ * @returns {Number}
+ */
+function getPoints() {
+  // Get the points
+  return backendProfile.value.getProfileData().rewardPoints;
+}
+
+/**
+ * Get the score from localStorage
+ *
+ * @returns {Number}
+ */
+function getScore() {
+  return backendProfile.value.getProfileData().score;
+}
+
+function addPoints(addedPoints) {
+  const profile = backendProfile.value.getProfileData();
+  profile.rewardPoints += addedPoints;
+  backendProfile.value.updateProfileData(profile);
+}
+
+function addScore(addedScore) {
+  const profile = backendProfile.value.getProfileData();
+  profile.score += addedScore;
+  backendProfile.value.updateProfileData(profile);
+}
+
+function completeTask(completed) {
```

Comment:
Display a notification for the user to indicate that they have acquired points etc